text/0040-token-metadata.md

```diff
@@ -0,0 +1,167 @@
+message TokenMetadata {
+    // The short name is a standard ticker symbol for the token
+    String short_name = 1;
+    // The long name is the standard english name for the token
+    String long_name = 2;
+    // The conversion factor is an unsigned 64 bit number expressed as a decimal string
+    String display_conversion_factor = 3;
+    // If present, a url pointing to a logo for the token
+    String logo_url = 4;
+    // If present, a url pointing to an MCIP which proposed the token
+    String mcip_url = 5;
+}
+```
+
+A consensus network client-facing API will be exposed to query for these objects given one or more token ids.
+
+The `tokens.toml` file schema is extended to contain these fields per token. The minting trust root should sign
+these fields.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This proposal
+
+* Adds to the complexity of configuring a network
+* Does not specify a mechanism that forces different nodes to have matching values
+  (we could choose to hash this data into responder id for instance)
+* Does not commit this data to the blockchain
+* Does not prevent the metadata from being changed -- only trust in the foundation not to sign another value
+* URL fields cannot be validated by users. For instance, we could include hash of the logo svg, and then they
+  could confirm the validity of the file they download.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The main rationale to justify the proposal is to try to make incremental progress increasing the discoverability
+of this important metadata.
+
+In the status quo, where different clients would have to hard-code this data, and then release it according to their
+release schedule, this may lead to a negative user experience. For example, if one user's wallet supports a new token id,
+and they decide to send it to another wallet of theirs that hasn't updated, or hasn't made a release that supports it yet,
+the new wallet will likely decide not to show this token to the user since it can't discover the metadata. From the user's
+point of view, their tokens have disappeared and they don't know how to get them back, so this is a negative user experience.
+
+With this proposal, all clients would be able to display all tokens with or without an update. So this seems to avoid
+a significant negative user experience.
+
+There are a number of ways this proposal could be strengthened however, described in the drawbacks section, around
+ensuring the integrity of this data.
+
+The main reasons not to do any of these is if we have uncertainty around whether / how this data should be prevented
+from being changed. For instance, it seems quite bad if the display conversion factor is changed after a token is launched.
+It might be desirable somehow for the logo or long name to be adjusted though.
+
+We could also allow this data to be committed to the chain, and for desired parts of it to be updateable, as part of a `MintConfigTx`
```

Comment:
FWIW I don't think it's particularly hard to add this extra data to the mint config tx, especially at this point in time where the new enclave has not yet been signed. It is also backwards-compatible since adding new fields would not break the proto.
The one disadvantage of adding it to the `MintConfigTx` is that it is more stuff to get correctly when issuing this transaction, a typo could very easily result in a lot of confusion to users. And realistically IMO the metadata is very unlikely to be changed over time.